formatter/table.go

```diff
@@ -0,0 +1,36 @@
+package formatter
+
+import (
+	"fmt"
+	"strings"
+	"unicode/utf8"
+)
+
+// formatTable formats a 2D string array into a table with aligned columns.
+// Each row must have the same number of columns.
+func formatTable(rows [][]string) string {
+	if len(rows) == 0 {
+		return ""
+	}
+
+	colWidths := make([]int, len(rows[0]))
+	for _, row := range rows {
+		for i, col := range row {
+			if w := utf8.RuneCountInString(col); w > colWidths[i] {
+				colWidths[i] = w
+			}
```

Comment:
So it's about this, no?

```suggestion
			colWidths[i] = max(colWidths[i], utf8.RuneCountInString(col))
```

Reply:
But I'm pretty sure slices.MaxFunc can help also, if you want to try

https://pkg.go.dev/slices#MaxFunc

Reply:
Thanks for review.

The 2 for loops can be replaced with the following using `MaxFunc`.

```
	for i := range colwidths {
		colwidths[i] = utf8.runecountinstring(slices.maxfunc(rows, func(row1, row2 []string) int {
			return utf8.runecountinstring(row1[i]) - utf8.runecountinstring(row2[i])
		})[i])
	}
```

What do you think?

Reply:
Let's wait for you to try `text/tabwriter` as @denisvmedia suggested.